src/app/index.scss

```diff
@@ -17,5 +17,49 @@
     width: 50px;
   }
 }
+
+.outer {
+  display: table;
+  position: absolute;
+  height: 100%;
+  width: 100%;
+}
+
+.middle {
+  display: table-cell;
+  vertical-align: middle;
+}
+
+.inner {
+  margin-left: auto;
+  margin-right: auto;
+}
```

Comment:
`outer`, `middle` and `inner` could go on a different file like `src/app/styles/vertical_rules.scss`